src/test/cypress/e2e/exercises/programming/ProgrammingExerciseParticipation.cy.ts

```diff
@@ -18,33 +20,79 @@ describe('Programming exercise participation', () => {
+            const submission = javaPartiallySuccessfulSubmission;
+            programmingExerciseEditor.makeSubmissionAndVerifyResults(exercise.id!, submission, () => {
+                programmingExerciseEditor.getResultScore().contains(submission.expectedResult).and('be.visible');
+            });
+        });
+
+        it('Makes a successful submission', () => {
+            programmingExerciseEditor.startParticipation(course.id!, exercise.id!, studentThree);
+            const submission = javaAllSuccessfulSubmission;
+            programmingExerciseEditor.makeSubmissionAndVerifyResults(exercise.id!, submission, () => {
+                programmingExerciseEditor.getResultScore().contains(submission.expectedResult).and('be.visible');
+            });
         });
     });
 
-    it('Makes a partially successful submission', () => {
-        programmingExerciseEditor.startParticipation(course.id!, exercise.id!, studentTwo);
-        const submission = partiallySuccessful;
-        programmingExerciseEditor.makeSubmissionAndVerifyResults(exercise.id!, submission, () => {
-            programmingExerciseEditor.getResultScore().contains(submission.expectedResult).and('be.visible');
+    describe('C programming exercise', () => {
+        let exercise: ProgrammingExercise;
+
+        before('Setup c programming exercise', () => {
+            cy.login(admin);
+            courseManagementRequest.createProgrammingExercise({ course, programmingLanguage: ProgrammingLanguage.C }).then((exerciseResponse) => {
+                exercise = exerciseResponse.body;
+            });
+        });
+
+        it('Makes a submission', () => {
+            programmingExerciseEditor.startParticipation(course.id!, exercise.id!, studentOne);
+            const submission = cAllSuccessful;
+            programmingExerciseEditor.makeSubmissionAndVerifyResults(exercise.id!, submission, () => {
+                programmingExerciseEditor.getResultScore().contains(submission.expectedResult).and('be.visible');
+            });
         });
     });
 
-    it('Makes a successful submission', () => {
-        programmingExerciseEditor.startParticipation(course.id!, exercise.id!, studentThree);
-        const submission = allSuccessful;
-        programmingExerciseEditor.makeSubmissionAndVerifyResults(exercise.id!, submission, () => {
-            programmingExerciseEditor.getResultScore().contains(submission.expectedResult).and('be.visible');
+    describe('Python programming exercise', () => {
+        let exercise: ProgrammingExercise;
+
+        before('Setup python programming exercise', () => {
+            cy.login(admin);
+            courseManagementRequest.createProgrammingExercise({ course, programmingLanguage: ProgrammingLanguage.PYTHON }).then((exerciseResponse) => {
+                exercise = exerciseResponse.body;
+            });
+        });
+
+        it('Makes a submission', () => {
```

Comment:
Why not add failing submissions also for Python and C?

Reply:
Because the programming exercises already take a lot of time (because we have to wait for the CI) and adding more exercises will again increase the CI time. With adding support for C and Python we just want to make sure, that the correct pipeline is selected on the CI and the results are returned correctly. Maybe this can be added in a follow-up in the future, if the CI gets faster.

Reply:
Understood, thanks